### PR TITLE
Add documentation for proof-of-possession access tokens

### PR DESCRIPTION
**Pull request by rohitnarula7176**
## Proposed changes

Describe what this PR is trying to do.
Add documentation for proof-of-possession access tokens

## Type of change

- [ ] Feature work
- [ ] Bug fix
- [x] Documentation
- [ ] Engineering change
- [ ] Test
- [ ] Logging/Telemetry

## Risk

- [ ] High – Errors could cause MAJOR regression of many scenarios. (Example: new large features or high level infrastructure changes)
- [ ] Medium – Errors could cause regression of 1 or more scenarios. (Example: somewhat complex bug fixes, small new features)
- [x] Small – No issues are expected. (Example: Very small bug fixes, string changes, or configuration settings changes)

### CODE REVIEW

File: docs/access_token-pop.md [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+   |        |<-(F)--- Protected Resource ---|               |
+   +--------+                               +---------------+
+   ```
+
+   2.1	In the Token Request, the client sends an authorization grant, e.g., an authorization code or a refresh token, to the authorization server in order to obtain an access token (and potentially a refresh token). The client proves the possession of a private key belonging to some public key by first generating an asymmetric RSA key pair on the device. The client then sends an additional parameter "req_cnf" to the token endpoint which contains a thumbprint of the RSA public key the client would like to associate with the access token. Please refer to public key thumbprint computation spec here [https://tools.ietf.org/id/draft-ietf-oauth-pop-key-distribution-04.html]()
+
+   2.2	The AS binds (sender-constrains) the access token to the public key claimed by the client; that is, the access token cannot be used without proving possession of the respective private key. This is signaled to the client by using the token_type value "pop" and by appending a "cnf" claim in the access token jwt containing a "kid" member identifying the public key.
```
Comment (oldalton) [inline docs/access_token-pop.md:35]: What is AS here? (might be worth explaining it in docs)

File: docs/access_token-pop.md [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+
+   ```objective-c
+   #if TARGET_OS_IPHONE
+       UIViewController *viewController = ...; // Pass a reference to the view controller that should be used when getting a token interactively
+       MSALWebviewParameters *webParameters = [[MSALWebviewParameters alloc] initWithAuthPresentationViewController:viewController];
+   #else
+       MSALWebviewParameters *webParameters = [MSALWebviewParameters new];
+   #endif 
+   
+   MSALInteractiveTokenParameters *interactiveParams = [[MSALInteractiveTokenParameters alloc] initWithScopes:scopes webviewParameters:webParameters];
```
Comment (oldalton) [inline docs/access_token-pop.md:67]: Do you need to set the auth scheme POP?

File: docs/access_token-pop.md [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+   ```
+   /**
+    Authentication Scheme to access the resource
+    */
+   @property (nonatomic, nullable) id<MSALAuthenticationSchemeProtocol> authenticationScheme;
+   ```
+
+   Create an instance of authenticationScheme which is either MSALAuthenticationSchemeBearer or MSALAuthenticationSchemePop. MSALAuthenticationSchemeBearer is the default authentication scheme used by the library for bearer access tokens and does not require explicit declaration.
+
+   MSALAuthenticationSchemePop has three required parameters,
```
Comment (oldalton) [inline docs/access_token-pop.md:97]: From below looks like it has 2 required parameters

File: docs/access_token-pop.md [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+   
+   /**
+    The authentication scheme for the tokens issued. For instance "Bearer " or "Pop".
+    */
+   @property (readonly, nonnull) NSString *authenticationScheme;
+   ```
+
+5. **Does MSAL still supports Bearer flows.**
+
+   Yes! PoP and Bearer flows may be used interchangeably with MSAL and with supported Authenticator versions **as long as the targeted resource supports it**.
```
Comment (oldalton) [inline docs/access_token-pop.md:162]: Let's also document supported Authenticator version?

**Review by oldalton — APPROVED** [review]

File: docs/access_token-pop.md (hunk 4 above)
Comment (rohitnarula7176) [reply docs/access_token-pop.md:162]: Addressed.

File: docs/access_token-pop.md (hunk 3 above)
Comment (rohitnarula7176) [reply docs/access_token-pop.md:97]: Good catch. Addressed.

File: docs/access_token-pop.md (hunk 2 above)
Comment (rohitnarula7176) [reply docs/access_token-pop.md:67]: It is explained below in step 5.3 of the doc.

File: docs/access_token-pop.md (hunk 1 above)
Comment (rohitnarula7176) [reply docs/access_token-pop.md:35]: AS is short hand for Authorization Server. Added explanation for that.

**Pull request merged**